File: ConexaoVagas API/Domains/BeneficioVaga.cs
using System;
using System.Collections.Generic;

namespace ConexaoVagasAPI.Domains
{
    public partial class BeneficioVaga
    {
        public int IdBeneficioVaga { get; set; }
        public int IdBeneficio { get; set; }
        public int IdVaga { get; set; }

        public Beneficio IdBeneficioNavigation { get; set; }
        public Vaga IdVagaNavigation { get; set; }
    }
}

File: ConexaoVagas Front/src/models/beneficioVaga.ts
import { Beneficio } from "./beneficio";
import { Vaga } from "./vaga";

export class BeneficioVaga {
    idBeneficioVaga?: number;
    idBeneficio?: number;
    idVaga?: number;
    idBeneficioNavigation?: Beneficio = undefined;
    idVagaNavigation?: Vaga = undefined;
}